src/binding/petsc4py/src/petsc4py/PETSc/DMComposite.pyx

```diff
@@ -1,31 +1,78 @@
 # --------------------------------------------------------------------
 
 cdef class DMComposite(DM):
+    """A DM object that is used to manage data for a collection of DMs."""
 
-    def create(self, comm=None):
+    def create(self, comm: Comm | None = None) -> Self:
+        """Create a composite object.
+
+        Collective.
+
+        Parameters
+        ----------
+        comm
+            The communicator of processors that will share the DM.
+
+        See also
+        --------
+        petsc.DMCompositeCreate
+
+        """
         cdef MPI_Comm ccomm = def_Comm(comm, PETSC_COMM_DEFAULT)
         cdef PetscDM newdm = NULL
         CHKERR( DMCompositeCreate(ccomm, &newdm) )
         PetscCLEAR(self.obj); self.dm = newdm
         return self
 
-    def addDM(self, DM dm, *args):
-        """Add DM to composite."""
+    # FIXME: API?
+    def addDM(self, DM dm, *args: DM) -> None:
+        """Add a DM vector to the composite.
+
+        Collective.
+
+        Parameters
+        ----------
+        dm
+            The DM object.
+        *args
+            Additional DM objects.
+
+        See also
+        --------
+        petsc.DMCompositeAddDM
+
+        """
         CHKERR( DMCompositeAddDM(self.dm, dm.dm) )
         cdef object item
         for item in args:
             dm = <DM?> item
             CHKERR( DMCompositeAddDM(self.dm, dm.dm) )
 
-    def getNumber(self):
-        """Get number of sub-DMs contained in the `DMComposite`."""
+    def getNumber(self) -> int:
+        """Get number of sub-DMs contained in the composite.
+
+        Not collective.
+
+        See also
+        --------
+        petsc.DMCompositeGetNumberDM
+
+        """
         cdef PetscInt n = 0
         CHKERR( DMCompositeGetNumberDM(self.dm, &n) )
         return toInt(n)
     getNumberDM = getNumber
 
-    def getEntries(self):
-        """Get tuple of sub-DMs contained in the `DMComposite`."""
+    def getEntries(self) -> tuple[DM, ...]:
```

Comment:
I think we should change this one to return `list[DM]`, and in all the other *FIXME* comments, we just keep returning lists.

*A foolish consistency is the hobgoblin of little minds.* -- R. W. Emerson.